src/org/jetbrains/plugins/scala/lang/psi/types/ScTypePsiTypeBridge.scala

```diff
@@ -169,6 +171,11 @@ trait ScTypePsiTypeBridge {
       case ScCompoundType(Seq(typez, _*), _, _) => toPsi(typez, project, scope)
       case ScDesignatorType(c: ScTypeDefinition) if ScType.baseTypesQualMap.contains(c.qualifiedName) =>
         toPsi(ScType.baseTypesQualMap.get(c.qualifiedName).get, project, scope, noPrimitives, skolemToWildcard)
+      case ScDesignatorType(valType: ScClass) if isValueType(valType) && !noPrimitives =>
```

Comment:
No primitives looks wrong.
This parameter required for parameterized type, so if ValueClass(OtherNormalClass) => Foo[ValueClass] goes to Foo[OtherNormalClass].